cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/BaseStorage.java

```diff
@@ -41,4 +43,37 @@ public Map<String, String> getProperties() {
         .map(HashMap::new)
         .orElseGet(HashMap::new);
   }
+
+  /**
+   * Allocates Table Space for the storage.
+   *
+   * <p>Default tableLocation looks like: {endpoint}/{rootPrefix}/{databaseId}/{tableId}-{tableUUID}
+   *
+   * @return the table location where the table data should be stored
+   */
+  @Override
+  public String allocateTableSpace(
```

Comment:
```suggestion
  public String allocateTableStorage(
```

Reply:
+1 I like the idea of "location" as a term.